### DISCUSSION TITLE
No SEO fields?

### DISCUSSION THREAD

**Discussion by rayjencode** [gatsbyjs/gatsby#32095] [Help]
Hi may I know how to get the SEO fields? thanks
![image](https://user-images.githubusercontent.com/3379074/113501055-4528e980-9555-11eb-8993-04de5902b641.png)

**Comment by rayjencode** [comment]
in the main graphql under shopify Im able to query it.

![image](https://user-images.githubusercontent.com/3379074/113502446-ecf6e500-955e-11eb-8467-54d003137bd5.png)

**Comment by smthomas** [comment]
This is a good catch and it's not very intuitive. I was able to get the seo field to show up by manually editing one of the SEO field data in Shopify. 

![image](https://user-images.githubusercontent.com/1393742/113521947-9390b500-9562-11eb-9ddc-ca0e0c3d449f.png)

Once I manually edited one of the products, that field now shows up:

![image](https://user-images.githubusercontent.com/1393742/113521962-ac996600-9562-11eb-8036-afa4e112c541.png)

This still seems like it might be a bug, though we will need to do some additional digging because this may be a problem with Shopify's admin API. It doesn't appear that Shopify is populating the default generated SEO data through the API.

**Comment by rayjencode** [comment]
@smthomas thanks for the reponse let me try this :)

**Comment by sslotsky** [comment]
This must be the result of using inference to determine which fields are defined on a node type. If there aren't any nodes with this SEO data, the schema doesn't know about it. We should consider being explicit with our schema field definitions rather than relying on inference, since that will add clarity to cases like these.

**Comment by rmcfadzean** [comment]
As a note, if you don't have the Online Store sales channel (aka you're using the "Shopify Lite" plan), I don't believe you can edit these fields without using the API.
I do agree that fixing the schema is a good idea though.

**Comment by dungle-scrubs** [comment]
Referring to what @smthomas mentioned, even you're being explicit with the schema, it looks like Shopify's API doesn't provide the data for the SEO fields unless it's been manually modified at least once (I've only tested this with a collection description).

The default SEO description will match what the shop owner adds to the normal description when the product is first created. Even though it exists, it's not pulled in through the API.
